webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -890,12 +890,29 @@ var userOverride = {
             // the browser call getVoices is async
             // so we are updating the array whenever they're available
             if (VOICE_OUTPUT_ENABLED && "speechSynthesis" in window) {
+                var isIosDevice = /iPhone|iPad|iPod/i.test(navigator.userAgent) || false;
+
                 AVAILABLE_VOICES_FOR_TTS = speechSynthesis.getVoices();
                 if (speechSynthesis.onvoiceschanged !== undefined) {
                     speechSynthesis.onvoiceschanged = function () {
                         AVAILABLE_VOICES_FOR_TTS = speechSynthesis.getVoices();
                     };
                   }
+                function hackForIosDevices() {
+                    /** 
+                        it is only for IOS devices so using newer syntax
+                        IOS devices would not let the speech API run programmatically unless we have triggered manually one time under the user's interaction.
+                     */
+
+                    const simulateFakeVoice = () => {
+                        const fakeInstance = new SpeechSynthesisUtterance('');
+                        speechSynthesis.speak(fakeInstance);
+                    };
+                    document.addEventListener('click', simulateFakeVoice, {
```

Comment:
unless a user clicks the, speech will not work right?

Reply:
yes, so I am considering the click-on Iframe document only once, so if a user initialize the widget we are good to go with TTS.
FYI: TTS is going to start work from welcome message itself